WWDC/SessionsTableViewController.swift

```diff
@@ -80,7 +83,7 @@ class SessionsTableViewController: NSViewController {
                 return
             }
 
-            let selectedRows = self.tableView.selectedRowIndexes.flatMap { (i) -> IndexedSessionRow? in
+            let selectedRows = selectedRowIndexes.flatMap { (i) -> IndexedSessionRow? in
```

Comment:
Moving this outside the lock is technically wrong. This code block can be move down inside the `sync` block, however, maintaining data integrity and calling the API on the main thread.

Reply:
Thanks for pointing that out. @antonselyanin can you change it following @allenhumphreys suggestion?

Reply:
Just a small follow up, it’s not just this code block, there are a couple pieces of logic that are only related to preserving selection that are scattered through the function. They can all be moved together. The only reason they’re separate was to make the code look pretty 🤓. I’d offer more specific guidance but I’m stuck with just my phone this weekend!

Reply:
I found another solution - I'll just disable Main Thread Checker for 'Test' mode and revert `SessionsTableViewController` changes.
'Main Thread Checked' will still be on for normal app runs.

It would be better to address `selectedRowIndexes` issue separately.